jiant/tasks/lm.py

```diff
@@ -69,7 +66,9 @@ def get_metrics(self, reset=False):
 
     def load_data(self):
         # Data is exposed as iterable: no preloading
-        pass
+        self.examples_by_split = {}
+        for split in self.files_by_split:
+            self.examples_by_split[split] = list(self.get_data_iter(self.files_by_split[split]))
```

Comment:
These changes will impact `AutoregressiveLanguageModelingTask`'s subclasses. Before this change data was "...exposed as iterable: no preloading", with this change it looks like calls to `load_data()` will store data in a new `self.examples_by_split` field in the Task objects. 

Can you please to confirm that this change is intended? If this change is necessary, with the final PR please provide confirmation that any performance changes are acceptable.

Reply:
Yes, this change is intended. Before, we weren't saving the data anywhere, and we were re-loading the data everytime we wanted to count the number of examples of get the sentences in the dataset to create the vocab.

Reply:
Yes, before your change the data was streamed by inheriting subclasses of `AutoregressiveLanguageModelingTask` (and like you say, it looks like it was streamed twice: once to count examples, and once to create the vocab).

I would assume however that the decision to stream the data was made to avoid loading the data into memory (and it also looks like the `pass`-logic that your change replaces was added in an effort to [delay data-loading](https://github.com/nyu-mll/jiant/pull/662)).

It looks to me like this change could cause a breakage (if the modified tasks have data that won't fit in memory — or if they're used in experiments where memory is already tight). So, if you want to make this change please confirm that the subclasses that inherit this behavior still work, and the change in memory footprint is acceptable.

**TL;DR** I wouldn't recommend making this change because it requires work to confirm that it's safe.

Reply:
Yeah, I guess it is a tradeoff between time and memory. Before I made this change, it took ~13 hours to do preprocessing, and making this change cut that to 1/3 the time. I see your point about memory though. 
In that case I'll just revert the change.